crates/ibc/src/core/ics02_client/handler/update_client.rs

```diff
@@ -1,19 +1,14 @@
 //! Protocol logic specific to processing ICS2 messages of type `MsgUpdateAnyClient`.
 
-use tracing::debug;
-
+use crate::core::ics02_client::error::ClientError;
 use crate::prelude::*;
 
-use crate::core::ics02_client::client_state::UpdatedState;
-use crate::core::ics02_client::error::ClientError;
-use crate::core::ics02_client::events::UpdateClient;
+use crate::core::ics02_client::events::{ClientMisbehaviour, UpdateClient};
 use crate::core::ics02_client::msgs::update_client::MsgUpdateClient;
 use crate::events::IbcEvent;
 
 use crate::core::context::ContextError;
 
-use crate::core::ics24_host::path::{ClientConsensusStatePath, ClientStatePath};
-
 use crate::core::{ExecutionContext, ValidationContext};
 
 pub(crate) fn validate<Ctx>(ctx: &Ctx, msg: MsgUpdateClient) -> Result<(), ContextError>
```

Comment:
> If there is a Tx that includes two update client messages what is the order for these function calls

It must always be 1, which is as you mentioned the behavior from dispatch. I added a [clarifying comment](https://github.com/cosmos/ibc-rs/pull/584/commits/478ed8ce12a4f2aba5f04facb49ec7dbfcfda1d3). This is also how Namada processes messages.

> I think the current equivalent of the ibc-go ValidateBasic() is in ibc-rs domain type conversion but we don't seem to do some of those checks there.

The goal is to do all validation in `validate()`, and no validation in `execute()`. If we put validation checks in domain type conversion, then these will be run also in `execute()`, which we're trying to avoid. I'd like us to move away from putting any checks in domain type conversions for this reason.